src/pacmod3_node.cpp

```diff
@@ -424,11 +238,356 @@ LNI::CallbackReturn PACMod3Node::on_error(const lc::State & state)
+      std::bind(&PACMod3Node::callback_rear_pass_door_cmd, this, std::placeholders::_1)),
+    std::shared_ptr<LockedData>(new LockedData(RearPassDoorCmdMsg::DATA_LENGTH)));
+  RCLCPP_INFO(this->get_logger(), "Initialized Japan Taxi-specific API");
+}
+
+void PACMod3Node::initializeVehicle4SpecificApi()
+{
+  can_pubs_[DetectedObjectRptMsg::CAN_ID] =
+    this->create_publisher<pacmod_msgs::msg::DetectedObjectRpt>(
+    "parsed_tx/detected_object_rpt", 20);
+  can_pubs_[DetectedObjectRptMsg::CAN_ID]->on_activate();
+
+  can_pubs_[VehicleDynamicsRptMsg::CAN_ID] =
+    this->create_publisher<pacmod_msgs::msg::VehicleDynamicsRpt>(
+    "parsed_tx/vehicle_dynamics_rpt", 20);
+  RCLCPP_INFO(this->get_logger(), "Initialized Vehicle4-specific API");
+}
+
+void PACMod3Node::initializeApiForMsg(uint32_t msg_can_id)
+{
+  // Need to initialize pubs/subs for this message group
+  switch (msg_can_id) {
+    case BrakeMotorRpt1Msg::CAN_ID:
+    case BrakeMotorRpt2Msg::CAN_ID:
+    case BrakeMotorRpt3Msg::CAN_ID:
+      {
+        initializeBrakeMotorRptApi();
+        break;
+      }
+    case SteerMotorRpt1Msg::CAN_ID:
+    case SteerMotorRpt2Msg::CAN_ID:
+    case SteerMotorRpt3Msg::CAN_ID:
+      {
+        initializeSteeringMotorRptApi();
+        break;
+      }
+    case WiperRptMsg::CAN_ID:
+    case WiperAuxRptMsg::CAN_ID:
+      {
+        initializeWiperApi();
+        break;
+      }
+    case HeadlightRptMsg::CAN_ID:
+    case HeadlightAuxRptMsg::CAN_ID:
+      {
+        initializeHeadlightApi();
+        break;
+      }
+    case HornRptMsg::CAN_ID:
+      {
+        initializeHornApi();
+        break;
+      }
+    case WheelSpeedRptMsg::CAN_ID:
+      {
+        initializeWheelSpeedApi();
+        break;
+      }
+    case ParkingBrakeRptMsg::CAN_ID:
+      {
+        initializeParkingBrakeRptApi();
+        break;
+      }
+    case DoorRptMsg::CAN_ID:
+      {
+        initializeDoorRptApi();
+        break;
+      }
+    case InteriorLightsRptMsg::CAN_ID:
+      {
+        initializeInteriorLightsRptApi();
+        break;
+      }
+    case OccupancyRptMsg::CAN_ID:
+      {
+        initializeOccupancyRptApi();
+        break;
+      }
+    case RearLightsRptMsg::CAN_ID:
+      {
+        initializeRearLightsRptApi();
+        break;
+      }
+    case HazardLightRptMsg::CAN_ID:
+      {
+        initializeHazardLightApi();
+        break;
+      }
+    case DateTimeRptMsg::CAN_ID:
+    case LatLonHeadingRptMsg::CAN_ID:
+    case YawRateRptMsg::CAN_ID:
+      {
+        initializeLexusSpecificApi();
+        break;
+      }
+    case CruiseControlButtonsRptMsg::CAN_ID:
+    case EngineBrakeRptMsg::CAN_ID:
+    case MarkerLampRptMsg::CAN_ID:
+    case SprayerRptMsg::CAN_ID:
+      {
+        initializeFreightlinerSpecificApi();
+        break;
+      }
+    case DetectedObjectRptMsg::CAN_ID:
+    case VehicleDynamicsRptMsg::CAN_ID:
+      {
+        initializeVehicle4SpecificApi();
+        break;
+      }
+  }
```

Comment:
Do we need an else to catch any mismatching CAN_ID?

Reply:
There's no functional need, as the `callback_can_tx` function will just skip/drop the CAN_ID that doesn't have a matching publisher. I don't want to add a ROS_WARN or anything either in case the customer is using a newer DBC or something, which could possibly result in constant ROS_WARN spam for CAN messages the driver doesn't support yet.